src/components/Settings/Security.js

```diff
@@ -0,0 +1,151 @@
+      .matches(
+        /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/,
+        INVALID_PASSWORD_FORMAT
+      )
+      .oneOf([yup.ref('newPassword'), null], 'Passwords do not match.'),
+  });
+  const { register, handleSubmit, errors, setError } = useForm({
+    revalidateMode: 'onChange',
+    resolver: yupResolver(passwordSchema),
+  });
+
+  const onSubmit = async data => {
+    try {
+      console.log(data);
+    } catch ({ data: { message } }) {
+      setError('email', {
+        type: 'manual',
+        message,
+      });
+    }
+  };
+
+  return (
+    <>
+      <form noValidate onSubmit={handleSubmit(onSubmit)}>
+        <div className={inputGroups}>
+          <h4>Current Password</h4>
+          <TextField
+            {...{
+              id: 'currentPassword',
+              className: `${
+                disabledInputs
+                  ? `${securityFields} ${disabledInputsBG}`
+                  : securityFields
+              }`,
+              variant: 'outlined',
+              inputProps: { 'data-testid': 'currentPassword' },
+              name: 'currentPassword',
+              type: 'password',
+              autoFocus: true,
+              inputRef: register,
+              disabled: disabledInputs,
+              error: !isEmpty(errors.currentPassword),
+              helperText:
+                errors.currentPassword && errors.currentPassword.message,
+            }}
+          />
+        </div>
+
+        <div className={inputGroups}>
+          <h4>New Password</h4>
+          <TextField
+            {...{
+              id: 'newPassword',
+              inputProps: { 'data-testid': 'newPassword' },
+              inputRef: register,
+              name: 'newPassword',
+              type: 'password',
+              variant: 'outlined',
+              disabled: disabledInputs,
+              className: `${
+                disabledInputs
+                  ? `${securityFields} ${disabledInputsBG}`
+                  : securityFields
+              }`,
+              error: !isEmpty(errors.newPassword),
+              helperText: errors.newPassword && errors.newPassword.message,
+            }}
+          />
+        </div>
+
+        <div className={inputGroups}>
+          <h4>Retype Password</h4>
+          <TextField
+            {...{
+              id: 'retypePassword',
+              inputProps: { 'data-testid': 'retypePassword' },
+              inputRef: register,
+              type: 'password',
+              name: 'retypePassword',
+              variant: 'outlined',
+              className: `${
+                disabledInputs
+                  ? `${securityFields} ${disabledInputsBG}`
+                  : securityFields
+              }`,
+              disabled: disabledInputs,
+              error: !isEmpty(errors.retypePassword),
+              helperText:
+                errors.retypePassword && errors.retypePassword.message,
+            }}
+          />
```

Comment:
seeing some repeating patterns that could be written in another way, will have to look into this

Reply:
I have removed the repetitive code by rendering html element through array

Reply:
resolved in commit # https://github.com/femmecubator/femmecubator-ui-repo/pull/150/commits/6317b65d216f29280546a4824e5ffd1cddb15629